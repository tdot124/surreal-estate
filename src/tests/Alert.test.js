import React from "react";
import { render, screen } from "@testing-library/react";
import Alert from "../components/Alert";

describe("Alert", () => {
  it("Displays an error message", () => {
    render(<Alert message="Error!" />);

    expect(screen.getByText(/Error/).textContent).toBe("Error!");
  });

  it("Displays a success message", () => {
    render(<Alert message="Success!!!" success />);
    expect(screen.getByText(/Success/).textContent).toBe("Success!!!");
  });

  it("Does not render when message prop is empty", () => {
    const { asFragment } = render(<Alert message="" />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("displays and has the correct class when success message passed", () => {
    const { asFragment } = render(<Alert message="Property Added" success />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("displays and has the correct class when error message passed", () => {
    const { asFragment } = render(
      <Alert message="Server Error. Please try again later." />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
