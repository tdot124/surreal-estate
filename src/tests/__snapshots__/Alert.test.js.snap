// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert Does not render when message prop is empty 1`] = `<DocumentFragment />`;

exports[`Alert displays and has the correct class when error message passed 1`] = `
<DocumentFragment>
  <div
    class="alert alert-error"
  >
    Server Error. Please try again later.
  </div>
</DocumentFragment>
`;

exports[`Alert displays and has the correct class when success message passed 1`] = `
<DocumentFragment>
  <div
    class="alert alert-success"
  >
    Property Added
  </div>
</DocumentFragment>
`;
